### ISSUE TITLE
Add "How to use custom gradients?" documentation

### ISSUE THREAD

**Issue by rlouf** [blackjax-devs/blackjax#318]
It's very simple: https://jax.readthedocs.io/en/latest/notebooks/Custom_derivative_rules_for_Python_code.html#python-debugging
